crates/editor/src/editor.rs

```diff
@@ -4064,34 +4063,24 @@ impl Editor {
         };
 
         self.signature_help_task = Some(cx.spawn(move |editor, mut cx| async move {
-            let markdown_task = editor
+            let signature_help_popover_task = editor
                 .update(&mut cx, |editor, cx| {
-                    let language = editor.language_at(position, cx);
```

Comment:
Hey @tomoikey, I've been following the development of this feature, thank you so much for this great contribution! :+1: Just wanted to pop in and say that I think what @SomeoneToIgnore meant in his original comment (https://github.com/zed-industries/zed/pull/12909#issuecomment-2199927205) about switching from markdown to rich text was to retain the language highlights. I'm not really very familiar with the Zed internals, but I think you could use `language.highlight_text` once you get the language at the cursor position. But I could be wrong about that, or perhaps it's not that easy, sorry in advance for any confusion on my part. :slightly_smiling_face:

Reply:
@filipwiech 
Nice to meet you! Thank you for your advice and suggestions for improvements! I tried using `language.highlight_text` in my environment, and it worked really well. However, upon reading the code more closely, I found that `parse_markdown` was already using it internally, so I indirectly adopted your suggestion. Previously, the highlight results were being discarded, so I modified the code to merge the results. Thanks to this, the signature_help_popover now includes language highlighting and looks much clearer. Thank you!

```rust
if let Some((markdown, mut markdown_highlights)) = markdown {
    let mut parsed_content =
        parse_markdown(markdown.as_str(), &language_registry, language).await;
    parsed_content.highlights.append(&mut markdown_highlights); // MERGE!!!
    Some(SignatureHelpPopover { parsed_content })
}
```